### PR TITLE
[Path] Fix open edge zero value start point

### PR DESCRIPTION
**Pull request by Russ4262**
forum mention at https://forum.freecadweb.org/viewtopic.php?f=15&t=64882
This fix applies a small value to start points for open edges when either the x or y start point value is roughly zero.

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [x]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [x]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [x]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [x]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the [FreeCAD bug tracker issue number](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) in case a particular commit solves or is related to an existing issue on the tracker. Ex: `Draft: fix typos - fixes #0004805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5343">pipeline status</a> for feature branch [PR_5343](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5343). Pipeline [443190374](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/443190374) was triggered at [9c83410](https://github.com/FreeCAD/FreeCAD/pull/5343/commits/9c834101c7dd28a65afe9591446c084095ccb464). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by benj5378** [comment]
The code looks fine. I haven't tested the functionality

File: src/Mod/Path/PathScripts/PathAreaOp.py [hunk 1]
```diff
@@ -269,11 +269,18 @@ def _buildProfileOpenEdges(self, obj, edgeList, isHole, start, getsim):
...
-                    lv = len(V) - 1
-                    pathParams['start'] = FreeCAD.Vector(V[0].X, V[0].Y, V[0].Z)
+                    verts = hWire.Wires[0].Vertexes
+                    idx = 0
                     if obj.Direction == 'CCW':
-                        pathParams['start'] = FreeCAD.Vector(V[lv].X, V[lv].Y, V[lv].Z)
+                        idx = len(verts) - 1
+                    x = verts[idx].X
+                    y = verts[idx].Y
+                    # Zero start value adjustments for Path.fromShapes() bug
```
[the remark this answers is not in the document]
Comment (agren) [reply src/Mod/Path/PathScripts/PathAreaOp.py:309]: Adding an image that hopefully clarifies what I try to say.
profile-results

[the remark this answers is not in the document]
Comment (Russ4262) [reply src/Mod/Path/PathScripts/PathAreaOp.py:323]: I grabbed a recent weekly build:
OS: Windows 10 Version 2009
Word size of FreeCAD: 64-bit
Version: 0.20.27428 (Git)
Build type: Release
Branch: master
Hash: 27460358508a2057e0ec57a418641435f12628dd
Python version: 3.8.6+
Qt version: 5.15.2
Coin version: 4.0.1
OCC version: 7.5.3
Locale: English/United States (en_US)

... and this is the result with what appears to be the same test file as you present above in the image.
Snip macro screenshot-f5e464

Applying the proposed fix in this PR corrects the path generated, as seen in the following image, and is applied in the following FreeCAD version:
OS: Windows 10 Version 2009
Word size of FreeCAD: 64-bit
Version: 0.20.27496 +2 (Git)
Build type: Release
Branch: fix/3D_pocket_overcut
Hash: f1a223ef64a3f29deeea8cb8d123997443a7f803
Python version: 3.8.6+
Qt version: 5.15.2
Coin version: 4.0.1
OCC version: 7.5.3
Locale: English/United States (en_US)

Snip macro screenshot-2e9f48

I have not looked at code for `Path.fromShapes()` to determine if a bug exists there.  As presented in the original test file by Lupin in the forum, the zero passed in the `pathParams["start"]` vector parameter are causing, or related to, the problem.  This PR intervenes within the python to make those zeros a very small non-zero number, which seems to correct the issue within the gcode output.

Thanks for reviewing and collaborating on this bug.

Russell

[the remark this answers is not in the document]
Comment (agren) [reply src/Mod/Path/PathScripts/PathAreaOp.py:323]: Sorry, I could have been clearer. I was not doubting your fix, it works. I was only doubting that the bug is in fromShapes. But I'm not so sure anymore.

Looking closer at the output generated by fromShapes does indeed show that the x feed is omitted when the X=0 in the start param.
Rapids between layers with model x=0:
```
Command G0 [ Z:5 ],
Command G0 [ Y:13 ],
Command G0 [ Z:3 ]
```
Rapids between layers with model x=1:
```
Command G0 [ Z:5 ],
Command G0 [ X:1 Y:13 ],
Command G0 [ Z:3 ]
```

So long story short. Ignore what I said :)

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5343">pipeline status</a> for feature branch [PR_5343](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5343). Pipeline [470441361](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/470441361) was triggered at [9227b4a](https://github.com/FreeCAD/FreeCAD/pull/5343/commits/9227b4ab2440f773f978f4987e839fba7bd99c96). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Pull request merged**